inc/saiswitch.h

```diff
@@ -1827,6 +1880,202 @@ typedef enum _sai_switch_attr_t
      */
     SAI_SWITCH_ATTR_NAT_ENABLE,
 
+    /**
+     * @brief SDK/Driver can access device through file system support.
+     *
+     * TRUE - Device access by file system.
+     * FALSE - Device access not supported directly by driver.
+     * Platform adaption device read/write API should be provided by the Host Adapter.
+     * @type bool
+     * @flags READ_ONLY
+     * @switchtype PHY
+     */
+    SAI_SWITCH_ATTR_HARDWARE_ACCESS_SUPPORT_BY_SYSFS,
+
+    /**
+     * @brief Switch hardware access bus MDIO/I2C/CPLD
+     *
+     * @type sai_switch_hardware_access_bus_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     * @default SAI_SWITCH_HARDWARE_ACCESS_BUS_MDIO
+     * @condition SAI_SWITCH_ATTR_HARDWARE_ACCESS_SUPPORT_BY_SYSFS == false
+     * @switchtype PHY
+     */
+    SAI_SWITCH_ATTR_HARDWARE_ACCESS_BUS,
+
+    /**
+     * @brief Platform context information
+     *
+     * Platform context information provided by the host adaptor to driver.
+     * This information is Host adaptor specific, typically used for maintain
+     * synchronization and device information. Driver will give this context back
+     * to adaptor as part of call back sai_switch_register_read/write_fn API.
+     *
+     * Use void as pointer.
+     *
+     * @type sai_pointer_t void
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     * @condition SAI_SWITCH_ATTR_HARDWARE_ACCESS_SUPPORT_BY_SYSFS == false
+     * @switchtype PHY
+     */
+    SAI_SWITCH_ATTR_PLATFROM_CONTEXT,
+
+    /**
+     * @brief Platform adaption device read callback function passed to the adaptor.
+     * This is mandatory function for driver when device access not supported by file system.
+     *
+     * Use sai_switch_register_read_fn as read function.
+     *
+     * @type sai_pointer_t sai_switch_register_read_fn
```

Comment:
currently all pointers declared in switch are considered notifications and must have _notification_t suffix, and _NOTIFY suffix on attribute name, an notifications functions must be declared void at this point.
This will pop up after you fix existing errors

Reply:
also im a little bit confused, why this function neds to be set here on attributes if we already have declared them in struct api:
 sai_switch_register_read_fn     switch_register_read;
 sai_switch_register_write_fn    switch_register_write;

Reply:
@kcudnik - Below 2 API's are given back by ASIC SAI implementation to provide PHY access by MDIO interface which is connected to NPU.
sai_switch_register_read_fn switch_register_read;
sai_switch_register_write_fn switch_register_write;

Platform adaption will have  has 2 parts.  PHY driver will call the call back API's to NOS, this is provided as part of NOTIFICATION attribute. After coming to NOS, in case PHY is connected through NPU then  NOS will call those API's to write data to PHY.

Reply:
please then follow convention for notification scheme as other pointers

Reply:
@AshokDaparthi Why those functions needs to be given back by SAI ?, you have declared those 2 functions in api structure, no need to have them as parameters, i would suggest to remove those or make those as notification
@lguohan Lets have meeting to explain what those are needed for